python/eos/analysis_file_description.py

```diff
@@ -0,0 +1,281 @@
+    lambda_scale:float
+    type:str=field(repr=False, init=False, default="scale")
+
+@dataclass
+class UniformPriorDescription(Deserializable):
+    parameter:str
+    min:float
+    max:float
+    type:str=field(repr=False, init=False, default="uniform")
+
+@dataclass
+class ConstraintPriorDescription(Deserializable):
+     constraint:str
+
+@dataclass
+class PriorComponent(Deserializable):
+    name:str
+    descriptions:list
+
+    @classmethod
+    def from_dict(cls, **kwargs):
+        _kwargs = _copy.deepcopy(kwargs)
+        if "descriptions" in kwargs:
+            if "parameters" in kwargs:
+                eos.error(f'Both \'descriptions\' and \'parameters\' are provided for prior component \'{kwargs["name"]}\', ignoring legacy support for \'parameters\'')
+            _kwargs['descriptions'] = [PriorDescription.from_dict(**d) for d in kwargs['descriptions']]
+        if "parameters" in kwargs:
+            eos.warn(f'\'parameters\' is in the description of prior component \'{kwargs["name"]}\', use \'descriptions\' instead')
+            _kwargs.pop("parameters")
+            _kwargs['descriptions'] = [PriorDescription.from_dict(**d) for d in kwargs['parameters']]
+        return Deserializable.make(cls, **_kwargs)
+
+
+
+@dataclass
+class ConstraintLikelihoodDescription(Deserializable):
+    constraint:eos.QualifiedName
+
+@dataclass
+class ManualConstraintDescription(Deserializable):
+    name:eos.QualifiedName
+    info:dict
+
+@dataclass
+class PyHFConstraintDescription(Deserializable):
+    file:str
+    parameter_map:dict=field(default_factory=dict)
+
+@dataclass
+class LikelihoodComponent(Deserializable):
+    name:str
+    constraints:list=field(default_factory=list)
+    manual_constraints:list=field(default_factory=list)
+    pyhf:list=field(default_factory=list)
+
+    @classmethod
+    def from_dict(cls, **kwargs):
+        if not ('constraints' in kwargs or 'manual_constraints' in kwargs or 'pyhf' in kwargs):
+            raise ValueError('LikelihoodComponent must have at least one of constraints, manual_constraints, or pyhf')
+        _kwargs = _copy.deepcopy(kwargs)
+        if 'constraints' in kwargs:
+            _kwargs['constraints'] = [ConstraintLikelihoodDescription.from_dict(constraint=c) for c in kwargs['constraints']]
+        if 'manual_constraints' in kwargs:
+            _kwargs['manual_constraints'] = [ManualConstraintDescription.from_dict(name=n, info=d) for n, d in kwargs['manual_constraints'].items()]
+        if 'pyhf' in kwargs:
+            _kwargs['pyhf'] = PyHFConstraintDescription.from_dict(**kwargs['pyhf'])
+        return Deserializable.make(cls, **_kwargs)
+
+
+
+@dataclass
+class PosteriorDescription(Deserializable):
+    name:str
+    prior:list
+    likelihood:list
+    global_options:dict=field(default_factory=dict)
+    fixed_parameters:dict=field(default_factory=dict)
+
+
+
+@dataclass
+class ObservableComponent(Deserializable):
+    name:str
+    latex:str
+    unit:str
+    options:dict
```

Comment:
``options`` should be optional

Reply:
Just FYI, in the current code, `options` is mandatory (see below) 
https://github.com/eos/eos/blob/e128cf93434ddb0378f98b3fee615eceb4c2f8dd/python/eos/analysis_file.py#L68-L77
But it does seem like it could be optional, since observables don't need to take options